Python 3.10 cffi build fails

<!--
The issue tracker is a tool to address bugs NOT a support platform.
Please use the Discord community or Stack Overflow for support questions,
more information at https://github.com/kivy/python-for-android#support
-->

### Checklist

- [ x] the issue is indeed a bug and not a support request
- [ x] issue doesn't already exist: https://github.com/kivy/python-for-android/issues
- [ x] I have a short, runnable example that reproduces the issue
- [x ] I reproduced the problem with the latest development version (`p4a.branch = develop`)
- [ x] I used the grave accent (aka backticks) to format code or logs when appropriated

### Versions

- Python: 2.10.6 (OS)
- OS: Ubuntu 22.04.2
- Kivy: 2.2.0
- Cython: 0.29.33
- OpenJDK:

### Description

p4a v2023.5.21  build of cffi fails

### buildozer.spec


Spec file:
```
requirements = python3,kivy, cffi
```

### Logs

```
/home/bobf/.buildozer/android/platform/android-ndk-r25b/toolchains/llvm/prebuilt/linux-x86_64/bin/clang -target aarch64-linux-android21 -fomit-frame-pointer -march=armv8-a -fPIC -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -target aarch64-linux-android21 -fomit-frame-pointer -march=armv8-a -fPIC -I/home/bobf/ex/hello/.buildozer/android/platform/build-arm64-v8a/build/other_builds/libffi/arm64-v8a__ndk_target_21/libffi/include -I/home/bobf/ex/hello/.buildozer/android/platform/build-arm64-v8a/build/other_builds/python3/arm64-v8a__ndk_target_21/python3/Include -DANDROID -I/home/bobf/.buildozer/android/platform/android-ndk-r25b/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include -I/home/bobf/ex/hello/.buildozer/android/platform/build-arm64-v8a/build/python-installs/apptest/arm64-v8a/include/python3.1 -fPIC -DUSE__THREAD -DHAVE_SYNC_SYNCHRONIZE -I/home/bobf/ex/hello/.buildozer/android/platform/build-arm64-v8a/build/other_builds/hostpython3/desktop/hostpython3/Include -I/home/bobf/ex/hello/.buildozer/android/platform/build-arm64-v8a/build/other_builds/hostpython3/desktop/hostpython3/native-build -c c/_cffi_backend.c -o build/temp.linux-x86_64-3.10/c/_cffi_backend.o
c/_cffi_backend.c:407:23: error: expression is not assignable
        Py_REFCNT(ct) = 43;
        ~~~~~~~~~~~~~ ^
c/_cffi_backend.c:410:23: error: expression is not assignable
        Py_REFCNT(ct) = 0;
        ~~~~~~~~~~~~~ ^

```

--- a/pythonforandroid/recipes/cffi/__init__.py
+++ b/pythonforandroid/recipes/cffi/__init__.py
@@ -7,7 +7,7 @@ class CffiRecipe(CompiledComponentsPythonRecipe):
     Extra system dependencies: autoconf, automake and libtool.
     """
     name = 'cffi'
-    version = '1.13.2'
+    version = '1.15.1'
     url = 'https://pypi.python.org/packages/source/c/cffi/cffi-{version}.tar.gz'
 
     depends = ['setuptools', 'pycparser', 'libffi']

--- a/pythonforandroid/recipes/cffi/disable-pkg-config.patch
+++ b/pythonforandroid/recipes/cffi/disable-pkg-config.patch
@@ -1,19 +1,19 @@
-diff --git a/setup.py b/setup.py
-index c1db368..57311c3 100644
+diff --git a/setup.py b/setup copy.py
+index 4ce0007..9be4a6d 100644
 --- a/setup.py
-+++ b/setup.py
-@@ -5,8 +5,7 @@ import errno
++++ b/setup
+@@ -9,8 +9,7 @@ if sys.platform == "win32":
  
  sources = ['c/_cffi_backend.c']
  libraries = ['ffi']
 -include_dirs = ['/usr/include/ffi',
 -                '/usr/include/libffi']    # may be changed by pkg-config
-+include_dirs = os.environ['FFI_INC'].split(",") if 'FFI_INC' in os.environ else []
- define_macros = []
++include_dirs = os.environ['FFI_INC'].split(',') if 'FFI_INC' in os.environ else []
+ define_macros = [('FFI_BUILDING', '1')]   # for linking with libffi static library
  library_dirs = []
  extra_compile_args = []
-@@ -67,14 +66,7 @@ def ask_supports_thread():
-         sys.stderr.write("The above error message can be safely ignored\n")
+@@ -105,14 +104,7 @@ def uses_msvc():
+     return config.try_compile('#ifndef _MSC_VER\n#error "not MSVC"\n#endif')
  
  def use_pkg_config():
 -    if sys.platform == 'darwin' and os.path.exists('/usr/local/bin/brew'):
@@ -25,6 +25,4 @@ index c1db368..57311c3 100644
 -    _ask_pkg_config(extra_link_args,    '--libs-only-other')
 -    _ask_pkg_config(libraries,          '--libs-only-l', '-l')
 +    pass
- 
- def use_homebrew_for_libffi():
-     # We can build by setting:
+

